dotcom-rendering/src/web/components/StickyVideo.importable.tsx

```diff
@@ -91,27 +92,64 @@ const stickyContainerStyles = (height: number) => css`
 interface Props {
 	isPlaying: boolean;
 	children: React.ReactNode;
+	videoId: string;
 }
 
-export const StickyVideo = ({ isPlaying, children }: Props) => {
+export const StickyVideo = ({ isPlaying, children, videoId }: Props) => {
 	const [isSticky, setIsSticky] = useState(false);
+	const [isClosed, setIsClosed] = useState(false);
+
 	const [isIntersecting, setRef] = useIsInView({
 		threshold: 0.5,
 		repeat: true,
 	});
 
 	useEffect(() => {
 		setIsSticky(isPlaying && !isIntersecting);
+		setIsClosed(false);
 	}, [isIntersecting, isPlaying]);
 
+	useEffect(() => {
+		if (isSticky) {
+			submitComponentEvent({
+				component: {
+					componentType: 'STICKY_VIDEO',
+					id: videoId,
+				},
+				action: 'STICK',
+			});
+		} else if (isClosed) {
+			submitComponentEvent({
+				component: {
+					componentType: 'STICKY_VIDEO',
+					id: videoId,
+				},
+				action: 'CLOSE',
+			});
+		} else {
+			submitComponentEvent({
+				component: {
+					componentType: 'STICKY_VIDEO',
+					id: videoId,
+				},
+				action: 'RETURN',
+			});
+		}
+	}, [isSticky, isClosed, videoId]);
+
+	if (!videoId) return null;
```

Comment:
Would this be better off at the top of the useEffect?

Reply:
I was at first trying to put it above the useEffect but I git lint error 🤔 
```
React Hook "useEffect" is called conditionally. React Hooks must be called in the exact same order in every component render
```
![image](https://user-images.githubusercontent.com/15894063/157217479-77423317-f929-4869-a20c-8d90fe150c76.png)

Reply:
Ah i meant inside the the useEffect!

```javascript
useEffect(() => {
  if(!videoId) return;
  [...]
}

```

Reply:
+1 to @joecowton1 's idea of putting it in the `useEffect` - I think the behaviour is the same either way but this feels closer to the intentions.

But.. is there maybe a way to make this prop required and to not even render the _component_ if it isn't. Then you wouldn't need to have this guard at all?

Reply:
Thanks both for your comments. I removed this check completely as the `videoId` is a required prop. But 1 question I have is that, although the prop is required but the videoId could still end up as an empty string, right?

Reply:
It could be an empty string, yes. You could check for this in the parent component but is this possibility, er, possible? Like, are we expecting this to happen?

Reply:
Perfect 👍